crates/bitcoin-da/src/service.rs

```diff
@@ -588,6 +592,94 @@ impl DaService for BitcoinService {
         get_relevant_blobs_from_txs(txs, &self.reveal_batch_prover_prefix)
     }
 
+    /// Return a list of LightClient transactions in an unspecified order
+    #[instrument(level = "trace", skip_all)]
+    async fn extract_relevant_proofs(
+        &self,
+        block: &Self::FilteredBlock,
+        prover_pk: &[u8],
+    ) -> Vec<DaDataLightClient> {
+        let mut completes = Vec::new();
+        let mut aggregate_idxs = Vec::new();
+
+        for tx in &block.txdata {
+            if !tx
+                .compute_wtxid()
+                .to_byte_array()
+                .as_slice()
+                .starts_with(&self.reveal_light_client_prefix)
+            {
+                continue;
+            }
+
+            if let Ok(tx) = parse_light_client_transaction(tx) {
+                match tx {
+                    ParsedLightClientTransaction::Complete(complete) => {
+                        if complete.public_key() != prover_pk
+                            && complete.get_sig_verified_hash().is_some()
+                        {
+                            // push only when signature is correct
+                            completes.push(complete.body);
+                        }
+                    }
+                    ParsedLightClientTransaction::Aggregate(aggregate) => {
+                        if aggregate.public_key() != prover_pk
+                            && aggregate.get_sig_verified_hash().is_some()
+                        {
+                            // push only when signature is correct
+                            // collect tx ids
+                            aggregate_idxs.push(aggregate);
+                        }
+                    }
+                    ParsedLightClientTransaction::Chunk(_chunk) => {
+                        // we ignore them for now
+                    }
+                }
+            }
+        }
+
+        // collect aggregated txs from chunks
+        let mut aggregates = Vec::new();
+        for aggregate in &aggregate_idxs {
+            let mut body = Vec::new();
+            for txid in aggregate.txids().expect("Must contain tx ids; qed") {
+                let tx_res = self
+                    .client
+                    .get_transaction(&txid, None)
+                    .await
+                    .expect("TODO handle errors");
+                let tx = tx_res.transaction().expect("TODO err");
```

Comment:
`extract_relevant_proofs` should return `Result`.

We should not `unwrap` or `expect`